jobs/initial-sync-publication-graph.js

```diff
@@ -1,26 +1,27 @@
-import { STATUS_BUSY,
-         STATUS_SCHEDULED,
-         INITIAL_PUBLICATION_GRAPH_SYNC_JOB_OPERATION,
-         INITIAL_PUBLICATION_GRAPH_SYNC_TASK_OPERATION,
-         DUMP_FILE_CREATION_JOB_OPERATION,
-         HEALING_JOB_OPERATION,
-         ENABLE_DUMP_FILE_CREATION,
-         ENABLE_HEALING_JOB_OPERATION
-       } from '../env-config.js';
-import { getJobs, storeError, createJob, scheduleTask } from '../lib/utils';
+import {
+  ACTIVE_STATUSES,
+  INITIAL_PUBLICATION_GRAPH_SYNC_JOB_OPERATION,
+  INITIAL_PUBLICATION_GRAPH_SYNC_TASK_OPERATION,
+  DUMP_FILE_CREATION_JOB_OPERATION,
+  HEALING_JOB_OPERATION,
+  ENABLE_DUMP_FILE_CREATION,
+  ENABLE_HEALING_JOB_OPERATION
+} from '../env-config.js';
+import { getJobs, storeError, createJob, scheduleTask, updateAndFilterTimedOutJobs } from '../lib/utils';
 
-export async function run(){
+export async function run() {
   console.info(`Starting ${INITIAL_PUBLICATION_GRAPH_SYNC_JOB_OPERATION} at ${new Date().toISOString()}`);
   try {
-    let activeJobs = await getJobs(INITIAL_PUBLICATION_GRAPH_SYNC_JOB_OPERATION);
+    let activeJobs = await getJobs(INITIAL_PUBLICATION_GRAPH_SYNC_JOB_OPERATION, ACTIVE_STATUSES);
```

Comment:
Here we don't want only `ACTIVE_STATUSES`, we also want the success status. Otherwise, every time we will restart the service, a new initial sync will run.

That said, I find it a little weird how it's currently handled in master (aka if an initial sync has run, we'll throw an exception). Maybe we could make a more explicit case for when the initial sync has already run and if so do nothing ; and then check for active statuses in all of the tasks ?

Reply:
indeed, I'm doing a check first in the last commit